charms/istio-pilot/src/charm.py

```diff
@@ -206,12 +206,48 @@ def install(self, _):
             f"values.global.proxy_init.image={global_proxy_init_image}",
         ]
 
+        if self._check_cni_configurations():
+            extra_flags.extend(
+                [
+                    "--set",
+                    "components.cni.enabled=true",
+                    "--set",
+                    f"values.cni.cniBinDir={self.model.config['cni-bin-dir']}",
+                    "--set",
+                    f"values.cni.cniConfDir={self.model.config['cni-conf-dir']}",
+                    "--set",
+                    "values.sidecarInjectorWebhook.injectedAnnotations.traffic\.sidecar\.istio\.io/excludeOutboundIPRanges=0.0.0.0/0",  # noqa
+                ]
+            )
+        return extra_flags
+
+    def install(self, _):
+        """Install charm."""
+        install_message = "Deploying Istio control plane with Istio CNI plugin."
+
+        # Verify the CNI configurations are set, otherwise we cannot continue with the
+        # Control Plane installation. Deferring the event is the only way we can prevent
+        # an attempt to install the Control Plane w/o these required values.
+        if not self._check_cni_configurations():
+            self.log.warning(
+                "This version of istio-pilot requires the cni-bin-dir and cni-conf-dir "
+                "configurations to be set in order to enable the Istio CNI plugin. "
+                "The installation with the Istio CNI plugin will be skipped. "
+                "If you want to enable this feature, please provide the required configuration."
+            )
+            install_message = (
+                "Deploying Istio control plane without Istio CNI plugin. "
+                "See juju-debug logs for more details."
+            )
+
```

Comment:
This warning feels out of place.  I think you're trying to recommend to people they should use the CNI plugin, but it says this version of istio-pilot requires CNI, but since you haven't asked for it we'll deploy without it.  "requires" in particular feels like the wrong word since we don't really require it

Reply:
Actually, what I am saying is that in order to enable the CNI plugin, this charm requires the cni configs; otherwise the plugin will be disabled. Is the message not clear bout it? I can re-phrase it if you think this should be different.

Reply:
Ah ok, then do we need these messages at all?  If this is an option that the user can enable, should the charm keep saying "you haven't enabled this option" on every reconcile?  That seems like a lot

The config and readme will say "you can enable CNI by setting these two configs, otherwise it'll use the standard version".  Isn't that enough notice to the user?  I can't think of other things in our charms we have that are optional but we keep reminding the user about like this

Reply:
Yeah, the more I think of this the more I agree. I think it is fair to assume people will know and they are informed about the type of installation they are receiving. Will remove in both install and upgrade methods.